proposals/xxxx-wave-matrix.md

```diff
@@ -0,0 +1,396 @@
+introduced in Shader Model 6.8.
+
+## Motivation
+
+GPUs have added dedicated hardware to support cooperative matrix multiplication
+across SIMD lanes. This feature proposes adding new data types and built-in
+functions to enable higher throughput matrix operations that fully utilize GPU
+SIMD hardware.
+
+These higher throughput matrix operations are required for optimal performance
+of many machine learning and image processing workloads. Adding native support
+to HLSL will enable high-performance matrix operations across all supported
+hardware with Shader Model 6.8 drivers.
+
+## Proposed solution
+
+WaveMatrix introduces new matrix templates to facilitate wave cooperative
+operations:
+
+```c++
+// Matrix Depth (K) dimension is hardware dependent
+// With TYPE_IN one of {float32_t, float16_t, uint8_t4_packed, int8_t4_packed}
+WaveMatrixLeft  <TYPE_IN, M, N> ;             // M x K
+WaveMatrixRight <TYPE_IN, M, N> ;             // K x N
+
+// With TYPE_ACC one of {float32_t, float16_t, int32_t}
+WaveMatrixAccumulator <TYPE_ACC, M, N> ;      // M x N
+// WaveMatrixLeftColAcc and WaveMatrixRightRowAcc are provided support for
+// quantization algorithms. See Zero Point section
+
+// For accumulating columns from WaveMatrixLeft into a single column of sums
+WaveMatrixLeftColAcc  <TYPE_ACC, M, N> ;      // M x 1
+
+// For accumulating rows from WaveMatrixRight into a single row of sums
+WaveMatrixRightRowAcc <TYPE_ACC, M, N> ;      // 1 x N
+```
+
+WaveMatrix accumulator object methods support operating on corresponding Left
+and Right operands. Results of operations can be stored or accumulated back into
+the accumulator. A simple example of multiplication is:
+
+```c++
+[numthreads(64,1,1)]
+void main(uint3 GTID : SV_GroupThreadID, uint GIDX : SV_GroupIndex)
+{
+  WaveMatrixLeft<float, 16, 16> Left;
+  WaveMatrixRight<float, 16, 16> Right;
+  WaveMatrixAccumulator<float, 16, 16> Acc;
+
+  Acc.Multiply(Left, Right); // Stores Left * Right into Acc
+  Acc.MultiplyAccumulate(Left, Right); // Adds Left * Right to Acc
+}
+```
+
+## Detailed design
+
+### Reading This Spec
+
+The next few sections include HLSL object definitions written in HLSL 2021
+syntax and using inheritance to represent interface composition. The objects in
+the `detail` namespace are not exposed in the HLSL runtime. They are provided
+here to make the specification more concise to consume. Objects in no namespace
+are exposed as public interfaces.
+
+### WaveMatrix Fill
+
+All WaveMatrix objects have a `Fill` method of the form `void Fill(ElTy Value)`
+where `ElTy` is the element type.
+
+The `Fill` method fills the matrix or matrix fragment with the provided value.
+All wave threads must provide the same value or the result is undefined. All
+WaveMatrix objects have the same `Fill` method with the same behavior.
+
+### WaveMatrix Matrix Objects
+
+The code below approximately declares the base interface that WaveMatrix matrix
+objects implement.

```

Comment:
Maybe add "the following sections will explain in detail what these methods do and what the parameters represent."